### PR TITLE
MOB-601 : Position Card Updates

### PR DESCRIPTION
**Pull request by mike-dydx**
## Links (dYdX Internal Use Only)
Linear Ticket: [MOB-601 : Position Card Updates](https://linear.app/dydx/issue/MOB-601/position-card-updates)
Figma Design: https://www.figma.com/design/mKevZOfE9nj6MZpiolKYW1/dYdX-%E2%80%BA-Mobile?node-id=8654-3911&t=JinIZ2gSpoIFSaNj-4

<br/>

## Description / Intuition
- adds notional value to position card
- cleans up position card layout. Previously hard coding some heights/widths

<br/>

## Before/After Screenshots or Videos

| Before | After |
|--------|-------|
|  |  |

Screenshot 2024-07-02 at 1 17 21 PM

<br/>

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring or Technical Debt
- [ ] Documentation update
- [ ] Other (please describe: )

### CODE REVIEW

File: dydx/dydxPresenters/dydxPresenters/_v4/Receipt/dydxTradeReceiptPresenter.swift [hunk 1]
```diff
@@ -43,17 +43,16 @@ final class dydxTradeReceiptPresenter: dydxReceiptPresenter {
...
+                let tradeSummary = input.summary
+                let marketId = input.marketId
+                let market = marketMap[marketId ?? ""]
+                let position = positions.first(where: { $0.id == marketId })
+                self?.updateExpectedPrice(tradeSummary: tradeSummary, market: market)
+                self?.updateLiquidationPrice(position: position, market: market)
+                self?.updatePositionMargin(position: position)
+                self?.updatePositionLeverage(position: position)
+                self?.updateTradingFee(tradeSummary: tradeSummary)
+                self?.updateTradingRewards(tradeSummary: tradeSummary)
```
Comment (mike-dydx) [inline dydx/dydxPresenters/dydxPresenters/_v4/Receipt/dydxTradeReceiptPresenter.swift:55]: this change ensures the initial (empty) state loads

despite null values, we still want to display the receipt line titles

File: dydx/dydxPresenters/dydxPresenters/_v4/Portfolio/Components/dydxPortfolioPositionsViewPresenter.swift [hunk 2]
```diff
@@ -119,7 +119,9 @@ class dydxPortfolioPositionsViewPresenter: HostedViewPresenter<dydxPortfolioPosi
         let item = positionsCache?[position.assetId] ?? dydxPortfolioPositionItemViewModel()
 
         let positionSize = abs(position.size.current?.doubleValue ?? 0)
+        let notionalValue = abs(position.valueTotal.current?.doubleValue ?? 0)
```
Comment (prashanDYDX) [inline dydx/dydxPresenters/dydxPresenters/_v4/Portfolio/Components/dydxPortfolioPositionsViewPresenter.swift]: hmmm i'm using `notionalTotal` here, what is `valueTotal`?

Comment (prashanDYDX) [reply dydx/dydxPresenters/dydxPresenters/_v4/Portfolio/Components/dydxPortfolioPositionsViewPresenter.swift]:
```
val valueTotal = size * oraclePrice
set(valueTotal, modified, "valueTotal", period)
val notional = valueTotal.abs()
set(notional, modified, "notionalTotal", period)
```

i think we want `notionalTotal` here

Comment (mike-dydx) [reply dydx/dydxPresenters/dydxPresenters/_v4/Portfolio/Components/dydxPortfolioPositionsViewPresenter.swift:122]: mm good catch. I was looking for notionalTotal earlier but autofill wasn't filling.

for our knowledge 
```
                            val valueTotal = size * oraclePrice
                            set(valueTotal, modified, "valueTotal", period)
                            val notional = valueTotal.abs()
                            set(notional, modified, "notionalTotal", period)
```

 addressed in

Comment (mike-dydx) [reply dydx/dydxPresenters/dydxPresenters/_v4/Portfolio/Components/dydxPortfolioPositionsViewPresenter.swift]: mmm good catch, i was looking for notionalTotal earlier but it wasn't autopopulating addressed in ba3fd6ca4bffc71d425d4bb79fc1300d881fd39b

**Review by prashanDYDX — APPROVED** [review]

**Pull request merged**